src/index.ts

```diff
@@ -378,109 +477,66 @@ export function postProcessConvertedFromZigbeeMessage(definition: Definition, pa
             try {
                 payload[key] = utils.calibrateAndPrecisionRoundOptions(value, options, expose.name);
             } catch (error) {
-                logger.logger.error(`Failed to apply calibration to '${expose.name}': ${(error as Error).message}`, NS);
-            }
-        }
-    }
-}
-
-export function removeExternalDefinitions(converterName?: string): void {
-    for (let i = 0; i < externalDefinitionsCount; i++) {
-        const definition = definitions[i];
-
-        if (converterName && definition.externalConverterName !== converterName) {
-            continue;
-        }
-
-        if ('zigbeeModel' in definition && definition.zigbeeModel) {
-            for (const zigbeeModel of definition.zigbeeModel) {
-                removeFromLookup(zigbeeModel, definition);
+                logger.error(`Failed to apply calibration to '${expose.name}': ${(error as Error).message}`, NS);
             }
         }
-
-        if ('fingerprint' in definition && definition.fingerprint) {
-            for (const fingerprint of definition.fingerprint) {
-                removeFromLookup(fingerprint.modelID, definition);
-            }
-        }
-
-        definitions.splice(i, 1);
-
-        externalDefinitionsCount--;
-        i--;
-    }
-}
-
-export function addDefinition(definition: DefinitionWithExtend): void {
-    definition = prepareDefinition(definition);
-
-    definitions.splice(0, 0, definition);
-
-    if (definition.externalConverterName) {
-        externalDefinitionsCount++;
-    }
-
-    if ('fingerprint' in definition && definition.fingerprint) {
-        for (const fingerprint of definition.fingerprint) {
-            addToLookup(fingerprint.modelID, definition);
-        }
-    }
-
-    if ('zigbeeModel' in definition && definition.zigbeeModel) {
-        for (const zigbeeModel of definition.zigbeeModel) {
-            addToLookup(zigbeeModel, definition);
-        }
     }
 }
 
-for (const definition of allDefinitions) {
-    addDefinition(definition as DefinitionWithExtend);
-}
-
 export async function findByDevice(device: Zh.Device, generateForUnknown: boolean = false): Promise<Definition | undefined> {
     let definition = await findDefinition(device, generateForUnknown);
 
-    if (definition && definition.whiteLabel) {
-        const match = definition.whiteLabel.find((w) => 'fingerprint' in w && w.fingerprint.find((f) => isFingerprintMatch(f, device)));
-
-        if (match) {
-            definition = {
-                ...definition,
-                model: match.model,
-                vendor: match.vendor,
-                description: match.description || definition.description,
-            };
+    if (definition) {
+        if (definition.whiteLabel) {
```

Comment:
```suggestion
        if (definition?.whiteLabel) {
```

nit: can be folder to one if level

Reply:
The return is outside the inner if, so it's needed (else would need a ternary on return, this way undefined is inferred).